pdns/dnsdist-lua-rules.cc

```diff
@@ -738,6 +739,52 @@ class RCodeRule : public DNSRule
   int d_rcode;
 };
 
+class ERCodeRule : public DNSRule
+{
+public:
+  ERCodeRule(int rcode) : d_rcode(rcode & 0xF), d_extrcode(rcode >> 4)
+  {
+  }
+  bool matches(const DNSQuestion* dq) const override
+  {
+    // avoid parsing EDNS OPT RR when not needed.
+    if (d_rcode != dq->dh->rcode) {
```

Comment:
Given that `dq->dh->rcode` is `unsigned`, perhaps `d_rcode` should be too? Same for `d_extrcode` since `edns0.extRCode` is an `uint8_t`.